### PR TITLE
write batch with capacity

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
@siddontang @BusyJay PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
I suggest doing a benchmark and profile the memory allocation cost.

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -56,6 +56,7 @@ use super::metrics::*;
 use super::local_metrics::{RaftReadyMetrics, RaftMessageMetrics, RaftProposeMetrics, RaftMetrics};
 
 const TRANSFER_LEADER_ALLOW_LOG_LAG: u64 = 10;
+const DEFAULT_APPEND_WB_SIZE: usize = 32 * 1024;
```
Comment (BusyJay) [inline src/raftstore/store/peer.rs]: Why `32 * 1024`?

**Comment by zhangjinpeng1987** [comment]
test writebatch::bench_writebatch::bench_writebatch_with_capacity    ... bench:       6,117 ns/iter (+/- 3,253)
test writebatch::bench_writebatch::bench_writebatch_without_capacity ... bench:       7,501 ns/iter (+/- 3,714)

**Comment by zhangjinpeng1987** [comment]
PTAL

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**